docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md

```diff
@@ -0,0 +1,177 @@
+- Multi-tenancy support at the namespace level can be enforced by the controller
+- When sealing secrets developers needs a connection to the cluster control plane to fetch the public key or the public key has to be explicitly shared with the developer
+- Potential risk of secret checked into a git repo
+- If the private key in the cluster is lost for some reason all secrets need to be re-encrypted followed by a new key-pair generation
+- Does not scale with Multi-cluster here every cluster will require a controller having its own key pair
+- Secrets are stored encrypted in the gitops repository, if the private encryption key is leaked, all secrets can be decrypted
+
+### [Mozilla SOPS](https://github.com/mozilla/sops)
+
+SOPS: Secrets OPerationS is an encryption tool that supports YAML, JSON, ENV, INI, and BINARY formats and encrypts with AWS KMS, GCP KMS, Azure Key Vault, age, and PGP and is not just limited to Kubernetes. It supports integration with some common key management systems including Azure Key Vault. Where the key management system is used to store the encryption key for encrypting secrets and not the actual secrets.
+
+- [Flux](https://toolkit.fluxcd.io/guides/mozilla-sops/#azure) has better support for SOPS with cluster-side decryption.
+- An added layer of security as the private key used for decryption is protected in an external key vault.
+- Needs plugin to work with Helm([Helm Secrets](https://github.com/jkroepke/helm-secrets)) and Kustomization ([KSOPS](https://github.com/viaduct-ai/kustomize-sops))([kustomize-sopssecretgenerator](https://github.com/goabout/kustomize-sopssecretgenerator))
+- Does not scale with larger teams as each developer has to encrypt the secrets
+- The public key is sufficient for creating brand new files. The secret key is required for decrypting and editing existing files because SOPS computes a MAC on all values.  When using the public key solely to add or remove a field, the whole file should be deleted and recreated.
+
+## 2. Reference to secrets stored in an external key vault (recommended)
+
+This approach relies on a key management system like [Azure Key Vault](https://learn.microsoft.com/en-us/azure/key-vault/general/overview) to hold the secrets and the git manifest in the repositories has reference to the key vault secrets. Developers do not perform any cryptographic operations with files in repositories. Kubernetes operators running in the target cluster are responsible for pulling the secrets from the key vault and making them available either as Kubernetes secrets or secrets volume mounted to the pod.
+
+For secret rotation ideas, see [Secrets Rotation on Environment Variables and Mounted Secrets](#secrets-rotation-on-environment-variables-and-mounted-secrets)
+
+For how to authenticate private container registries with a Service principal see: [Authenticated Private Container Registry](#authenticated-private-container-registry)
```

Comment:
lower caps for "Service principal" --> "service principal". Might be better to call it an "Azure Active Directory (Azure AD) service principal".